scalameta/common/shared/src/main/scala/org/scalameta/adt/Reflection.scala

```diff
@@ -121,7 +121,8 @@ trait Reflection {
     override def toString = s"leaf $prefix"
   }
   class Field(val sym: Symbol) {
-    if (!sym.isField) sys.error(s"$sym is not a field")
+    if (!sym.isField && !sym.hasAnnotation[AstMetadata.binaryCompatField])
```

Comment:
This is added so that we can represent binary compat values as fields and mostly need in one place.